src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -149,53 +143,32 @@ void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
 				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
 	}
 
+	is_updated = dim_feature_space == dim;
+	SG_DEBUG("set_dim_feature_space(), is_updated = {}",is_updated);
 	dim_feature_space = dim;
 
 }
 
 int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
+	return random_weights.num_cols;
 }
 
 void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
 	if (kernelwidth2 <= 0) {
 		throw ShogunException(
 				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
 	}
+
+	is_updated = kernel_width == kernelwidth2;
```

Comment:
I dont like this "updated" thing. 
I'd just remove all of that tbh. If a user hits `fit` then the thing is fitted/computed. Otherwise nothing happens. Changing the feature space dim after it has been fitted just results in nothing ... until the next time fit is called

Reply:
and all the check whether things are internally consistent can happen inside "fit"

Reply:
> I dont like this "updated" thing.
> I'd just remove all of that tbh. If a user hits `fit` then the thing is fitted/computed. Otherwise, nothing happens. Changing the feature space dim after it has been fitted just results in nothing ... until the next time fit is called

It's...a necessary evil. Thing is, we want to avoid recomputing if feature_dim_space and input_dim_space hasn't changed. We need something to keep track of this, I see your point though. I'll do better.

Reply:
I think we should just remove this whole state story.
`fit` instantiates the random basis with whatever values are present.
`apply` ( or the like ) simply project based on that (i.e. they just use the dimensions of the basis, not the parameters). If no basis is set, error. Easy

Reply:
actually, we don't even need to store the input dimensions. That can be extracted from the features upon `fit`. The only parameter should be the kernel (representation, in this case the bandwidth -- which should be refactored in the future), and the number of feature features

Reply:
> actually, we don't even need to store the input dimensions. That can be extracted from the features upon `fit`. The only parameter should be the kernel (representation, in this case the bandwidth -- which should be refactored in the future), and the number of feature features

Oh we're past this already :) . I've removed input_dim_features , get_dim_input_features() just returns m_random_weights.num_cols.